### DISCUSSION TITLE
Typescript AxiosError response.data의 내부 type 지정 관련 질문

### DISCUSSION THREAD

**Discussion by HyeongtaekOh** [wafflestudio/seminar-2021#636] [Q&A] [answered]
axios를 이용하는 함수의 오류처리 catch문에서 API 서버에서 돌아오는 error message를 기반으로 에러 토스트를 띄우고 있는데 해당 부분을 Typescript로 바꾸려하니 아래처럼 error response의 data type이 지정되어있지 않아 오류가 뜨는데 AxiosError의 내부 type을 지정할 수 있는 방법이 있나요?
![스크린샷, 2021-11-05 19-14-24](https://user-images.githubusercontent.com/86342741/140495741-83113d77-4311-49dc-9d0c-34feebb8f9b4.png)

**Comment by woohm402** [comment, marked as answer]
이전가지의 해결방법과 마찬가지로 [해당 부분 소스](https://github.com/axios/axios/blob/c5fe05bdff82e0fa4ace9a832e75052e1ee297f0/index.d.ts#L124-L131) 를 보면 generic을 이용해 어떻게 잘 구현되어 있단 걸 확인할 수 있습니다.

```ts
e as AxiosError<뭔가>
```

로 `T`를 잡아주시면 됩니다.

**Reply by woohm402** [reply]
사실 매번 소스를 찾아보긴 귀찮으니, 웹스톰 사용중이신 것 같은데 `ctrl` (또는 `command`) + 클릭 하시면 `node_modules` 안에 있는 해당 파일로 직접 가보실 수 있습니다!

**Reply by woohm402** [reply]
https://user-images.githubusercontent.com/39977696/140522454-edf15947-69c6-42fa-b38f-d0674ee23b5b.mov

**Reply by HyeongtaekOh** [reply]
해보겠습니다!

**Reply by HyeongtaekOh** [reply]
해결됐습니다. 비슷한 부분 가지고 계속 헤맸네요...ㅎ
감사합니다!
